### PR TITLE
Cronless Crons, Staff Roles

### PR DESCRIPTION
**Pull request by Orsokuma**
And a few fixes

### CODE REVIEW

File: installer.php [hunk 1]
```diff
@@ -513,11 +513,13 @@ function install(): void
...
-    * * * * * php $path/cron_minute.php $code
-    0 * * * * php $path/cron_hour.php $code
-    0 0 * * * php $path/cron_day.php $code
-    </pre>
+    * * * * * php $path/crons/CronHandler.php cron=minute-1 code=$code
+    */5 * * * * php $path/crons/CronHandler.php cron=minute-5 code=$code
+    0 * * * * php $path/crons/CronHandler.php cron=hour-1 code=$code
+    0 0 * * * php $path/crons/CronHandler.php cron=day-1 code=$code
+    </pre><br>
+    Alternatively, you can toggle the \"Use Timestamps Instead of Cron Jobs\" option in the Basic Settings on the Staff Panel to use timestamps instead.<br><br>
```
Comment (davemacaulay) [inline installer.php:521]: Is there a performance hit to whichever user runs this as part of request?

File: global_func.php [hunk 2]
```diff
@@ -1278,3 +1278,109 @@ function userBox(int|string $target_id): string
...
     return $target_id;
 }
+
+/**
+ * @param string|array $permissions
+ * @param bool $exit
+ * @param int|null $target_id
+ * @return bool|void
+ */
+function check_access(string|array $permissions, bool $exit = true, ?int $target_id = null)
```
Comment (davemacaulay) [inline global_func.php:1288]: This function has two unpredictable stats. This creates a confusing understanding of the usage

`check_access('test')` - will just exit, which cannot be represented easily in the editor and will look like the rest of the code will just run.
`check_access('test', false)` - returns an assertion based on the required parameters.

I would promote dropping the `bool $exit` parameter and always have this return a bool, and handle the eventual operation in the usage for better understandable code.

File: globals.php [hunk 3]
```diff
@@ -56,6 +56,10 @@
 $db->connect();
 $c = $db->connection_id;
 $set = get_site_settings();
+if ($set['use_timestamps_over_crons']) {
+    define('SILENT_CRONS', true);
+    require_once __DIR__ . '/crons/cronless_crons.php';
```
Comment (davemacaulay) [inline globals.php:61]: Even though the eventual `run` call is error handled, I think there'd be value of try catching this whole require as to not blow up a users page if something else were to fail

**Review by davemacaulay — CHANGES_REQUESTED** [review]
Few minor comments which I think can help code understanding

File: installer.php (hunk 1 above)
Comment (Orsokuma) [reply installer.php:507]: There will likely be an exponential performance hit as the database grows bigger.
This method was added more as a stop-gap measure for a lack of cron job support on a per-(local)host situation.

Production-ready games with cron support are advised to make use of said cron jobs and not rely on this in the long-term.

File: global_func.php (hunk 2 above)
Comment (Orsokuma) [reply global_func.php]: Resolved by [af2ec41760a69976afcc9f7fd41548392cb4e30e](https://github.com/davemacaulay/mccodesv2/commit/af2ec41760a69976afcc9f7fd41548392cb4e30e)

File: globals.php (hunk 3 above)
Comment (Orsokuma) [reply globals.php]: Addressed by [969d72383f9e0366b3e6a226e9ef0450831b21e0](https://github.com/Orsokuma/mccodesv2-php8/commit/969d72383f9e0366b3e6a226e9ef0450831b21e0)

**Pull request merged**